src/pages/ScreenRouting/ScreenRouter.tsx

```diff
@@ -0,0 +1,85 @@
+import { useParams } from 'react-router';
+import { useTranslation } from 'react-i18next';
+import React from 'react';
+
+import useMedia from '#src/hooks/useMedia';
+import Loading from '#src/pages/Loading/Loading';
+import ErrorPage from '#src/components/ErrorPage/ErrorPage';
+import usePlaylist from '#src/hooks/usePlaylist';
+import type { Playlist, PlaylistItem } from '#types/playlist';
+import { isEpisode, isSeriesPlaceholder } from '#src/utils/media';
+import MediaMovie from '#src/pages/ScreenRouting/screens/MediaMovie/MediaMovie';
+import MediaSeries from '#src/pages/ScreenRouting/screens/MediaSeries/MediaSeries';
+import MediaSeriesEpisode from '#src/pages/ScreenRouting/screens/MediaSeriesEpisode/MediaSeriesEpisode';
+import MediaLiveChannel from '#src/pages/ScreenRouting/screens/MediaLiveChannel/MediaLiveChannel';
+import PlaylistGrid from '#src/pages/ScreenRouting/screens/PlaylistGrid/PlaylistGrid';
+import PlaylistLiveChannels from '#src/pages/ScreenRouting/screens/PlaylistLiveChannels/PlaylistLiveChannels';
+import { ScreenMap } from '#src/pages/ScreenRouting/ScreenMap';
+
+type ScreenType = 'media' | 'playlist';
+
+type Props = {
+  type: ScreenType;
+};
+
+export const mediaScreenMap = new ScreenMap<PlaylistItem>();
+export const playlistScreenMap = new ScreenMap<Playlist>();
+
+// register media screens
+mediaScreenMap.registerByContentType(MediaSeries, 'series');
+mediaScreenMap.registerByContentType(MediaSeriesEpisode, 'episode');
+mediaScreenMap.registerByContentType(MediaLiveChannel, 'livechannel');
+mediaScreenMap.registerDefault(MediaMovie);
+
+// register legacy series and episode screens when `contentType` is missing
+mediaScreenMap.register(MediaSeriesEpisode, (item) => !!item && isEpisode(item));
+mediaScreenMap.register(MediaSeries, (item) => !!item && isSeriesPlaceholder(item));
+
+// register playlist screens
+playlistScreenMap.registerByContentType(PlaylistLiveChannels, 'live');
+playlistScreenMap.registerDefault(PlaylistGrid);
+
+/**
+ * The ScreenRouter can be used as page in the main Router. It will fetch a media or playlist based on the `type`
+ * parameter and shows a basic error screen when the data doesn't exist.
+ *
+ * When the data is loaded, a Screen component is determined based on the `contentType`.
+ */
+const ScreenRouter: React.VFC<Props> = ({ type }) => {
```

Comment:
Would it make sense to create separate MediaScreenRouter and PlaylistScreenRouter?  There doesn't seem to be much duplication here.  My apologies if I suggested combining them in the first place...